examples/arp_packet.rs

```diff
@@ -0,0 +1,96 @@
+extern crate pnet;
+
+use std::env;
+use std::process;
+use std::io::{self, Write};
+use std::net::{Ipv4Addr, AddrParseError};
+
+use pnet::util::{MacAddr, ParseMacAddrErr};
+use pnet::datalink::{self, NetworkInterface};
+use pnet::datalink::Channel;
+use pnet::packet::ethernet::MutableEthernetPacket;
+use pnet::packet::arp::MutableArpPacket;
+use pnet::packet::ethernet::EtherTypes;
+use pnet::packet::MutablePacket;
+use pnet::packet::arp::{ArpHardwareTypes, ArpOperations, ArpOperation};
+
+
+
+fn send_arp_packet(interface: NetworkInterface, source_ip: Ipv4Addr, source_mac: MacAddr, target_ip: Ipv4Addr, target_mac: MacAddr, arp_operation: ArpOperation) {
+    let(mut tx, _) = match datalink::channel(&interface, Default::default()) {
+        Ok(Channel::Ethernet(tx, rx)) => (tx, rx),
+        Ok(_) => panic!("Unknown channel type"),
+        Err(e) => panic!("Error happened {}", e),
+    };
+
+    /// ethernet_packet = Ethernet {
+    ///     destination: [0xff, 0xff, 0xff, 0xff, 0xff, 0xff],
+    ///     source: [0x28, 0xef, 0xf9, 0x5f, 0x8e, 0x2b],
+    ///     ethertype: [0x08, 0x06], // Arp(0x0806)
+    ///     payload: arp_packet
+    /// }
+    ///
+    /// arp_packet = Arp {
+    ///     hardware_type: [0x00, 0x01],
+    ///     protocol_type: [0x08, 0x00], // Ipv4(0x0800)
+    ///     hw_addr_len: [0x06],
+    ///     proto_addr_len: [0x04],
+    ///     operation: [0x00, 0x02], // Reply(0x0002)
+    ///     sender_hw_addr: [0x28, 0xef, 0xf9, 0x5f, 0x8e, 0x2b],
+    ///     sender_proto_addr: [0xc0, 0xa8, 0x00, 0x66], // Ipv4(192.168.0.102)
+    ///     target_hw_addr: [0xff, 0xff, 0xff, 0xff, 0xff, 0xff], // Broadcast
+    ///     target_proto_addr: [0xc0, 0xa8, 0x00, 0x65], // Ipv4(192.168.0.101)
+    ///     payload: [],
+    /// }
+
+    let mut ethernet_buffer: Vec<u8> = (0..42).map(|_| 0x00).collect::<Vec<u8>>();
```

Comment:
Usually, for allocating packets, you would use:
```rust
let mut ethernet_buffer = [0u8; 42];
```
instead, which will allocate 42 bytes of zeroes on the stack.